src/components/MoviesList/MovieBox/MovieBox.jsx

```diff
@@ -1,24 +1,118 @@
+
+    showDropdown = () => {
+        this.setState({
+            showDropdown: true
+        })
+    };
+
+    hideDropdown = () => {
+        this.setState({
+            showDropdown: false
+        })
+    };
+
+    handleShowModalEdit = () => {
+      this.setState({
+          isShowModalEdit: true
+      })
+    };
+
+    handleShowModalDelete = () => {
+        this.setState({
+            isShowModalDelete: true
+        })
+    };
+
+    handleCloseModalDelete = () => {
+        this.setState({
+            isShowModalDelete: false
+        })
+    };
+
+    handleCloseModalEdit = () => {
+        this.setState({
+            isShowModalEdit: false
+        })
+    };
+
+    render() {
+        const {poster_path, title, genres, release_date} = this.props.movie;
+
+        return (
+            <>
+                <div className="movie-box" onMouseEnter={this.showButtonDropdown} onMouseLeave={this.hideButtonDropdown}>
+                    <img src={poster_path} alt=""/>
+                    <div className="movie-description">
+                        <div className="movie-heading">
+                            <h2>{title}</h2>
+                            <span className="movie-title">{genres.join(', ')}</span>
+                        </div>
+                        <span className="movie-year">{release_date.getFullYear()}</span>
+                    </div>
+
+                    {
+                        this.state.showButtonDropdown ? (
+                        <div className="movie-dropdown">
+                            <button type="button" onClick={this.showDropdown} className="movie-edit-icon"></button>
+                            {
+                                this.state.showDropdown ? (
+                                    <div className="movie-dropdown-wrap">
+                                        <MovieDropdownList
+                                            onModalEditClick={this.handleShowModalEdit}
+                                            onModalDeleteClick={ this.handleShowModalDelete} />
+                                        <button
+                                            className="movie-dropdown-close"
+                                            onClick={this.hideDropdown}
+                                        >X</button>
+                                    </div>
+                                ): null
+                            }
+                        </div>
+                        ): null
+                    }
                 </div>
-                <span className="movie-year">{release_date.getFullYear()}</span>
-            </div>
-        </div>
-    )
-};
+                <Modal isOpen={this.state.isShowModalEdit} onClose={this.handleCloseModalEdit}>
+                    <EditMovieForm />
+                </Modal>
+                <Modal isOpen={this.state.isShowModalDelete} onClose={this.handleCloseModalDelete}>
+                    <DeleteMovie />
+                </Modal>
```

Comment:
hm, from my point of view it's not a good idea for MovieBox component to know how many different modals exist. What if we pass down to Modal itself to decide which one to show? 
For example: you can have types of modals 
```
const modals = {
   EDIT: 'edit',
   ADD: 'add',
   DELETE: 'delete'
}
```
and pass this value down to `Modal` along with handler for it.

Thus, you'll have one `<Modal type={modals.EDIT} handler={() => {} /}>` here .

And avoid bunch of state booleans: `isShowModalEdit`, `isShowModalDelete`.
They can be replaced with one `modalType` (name can be changed) with default to `null` (since you don't want to show any modals initially).

Reply:
I just had a second thought, and I think with this approach that counter inside `Modal` component can be avoided, can't it?